consensus/consensusfsm/fsm.go

```diff
@@ -408,11 +408,13 @@ func (m *ConsensusFSM) prepare(evt fsm.Event) (fsm.State, error) {
 	}
 
 	overtime := m.ctx.WaitUntilRoundStart()
+	if proposal != nil {
```

Comment:
need to check `IsENode()` here? delegates don't need to broadcast proposal

Reply:
or as phase 1, ENode and CNode still use the same dpos code?
if that's the plan, need to add the `IsENode()` in the consensus/dpos module? since lots of code/funcs will only used by CNode